Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs

```diff
@@ -1,32 +1,52 @@
 ﻿using FastEndpoints;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Shared.DTOs;
 
 namespace Queueomatic.Server.Endpoints.User.Edit;
 
 public class EditUserEndpoint: Endpoint<EditUserRequest, EditUserResponse>
 {
+    private readonly IUnitOfWork _unitOfWork;
+
+    public EditUserEndpoint(IUnitOfWork unitOfWork)
+    {
+        _unitOfWork = unitOfWork;
+    }
     public override void Configure()
     {
-        Verbs(Http.PUT);
-        Routes("/users/{email}");
-        AllowAnonymous();
+        Put("/users/{email}/update/profile");
+        Roles("User");
+        Policies("SignedInUser");
     }
 
     public override async Task HandleAsync(EditUserRequest req, CancellationToken ct)
     {
-        try
+        if (!User.IsInRole("User") && !req.Email.Equals(req.UserId))
         {
-            var response = new EditUserResponse();
-            await SendAsync(response, 204, cancellation: ct);
+            await SendUnauthorizedAsync();
+            return;
         }
-        catch (NullReferenceException nullException)
+        var user = await _unitOfWork.UserRepository.GetAsync(req.Email);
+        if (user is not null)
         {
-            Logger.LogInformation($"The request can not be null.\nMessage: {nullException.Message}");
-            await SendAsync(response:null, 400, ct);
+            user.Email = req.User.Email;
+            user.NickName = req.User.NickName;
         }
-        catch (TaskCanceledException exception)
-            when(exception.CancellationToken == ct)
+
+        await _unitOfWork.UserRepository.UpdateAsync(user);
+        
+        var result = await _unitOfWork.SaveAsync();
+        
+        if (result == 0)
         {
-            Logger.LogInformation($"Task {nameof(EditUserEndpoint)} was cancelled.");
+            await SendErrorsAsync();
+            return;
         }
+
+        await SendAsync(new(new UserDto
```

Comment:
Checkout the API specification for the actual response code.

Reply:
what do you mean this?

Reply:
This endpoint should respond with no content.

Reply:
> Checkout the API specification for the actual response code.

Is the api specification not incorrect here? We are specifying a 204 which means no content, but you are supposed to return a user as response body. Are those two not contradicting?

Reply:
It can be both 200 and 204. Here makes changes and the response will not be used.